packages/mobile/src/screens/product-info/product-info.tsx

```diff
@@ -31,34 +39,70 @@ import {
 
 const ProductInfo: FC = () => {
   const { colors } = useCustomTheme();
+  const [isLoading, setIsLoading] = useState(false);
   const dispatch = useAppDispatch();
   const product = useAppSelector(selectProduct);
+  const user = useAppSelector(selectCurrentUser);
+  const favoriteIds = useAppSelector(selectFavoriteIds);
   const route =
     useRoute<
       RouteProp<Pick<RootNavigationParamList, RootScreenName.ITEM_INFO>>
     >();
   const id = route.params?.itemId;
+  const isFavorite = favoriteIds.includes(id);
 
   useEffect(() => {
     dispatch(productActions.loadProductInfo(id));
     dispatch(productActions.updateProductViews(id));
+    if (user) {
+      if (favoriteIds) {
+        favoriteIds.map((id) => dispatch(productsActions.addToFavorite(id)));
+        dispatch(productsActions.cleanFavoriteIds());
+      }
+      dispatch(productsActions.fetchFavoriteIds());
+    }
   }, []);
 
   if (!product) {
     return <Spinner />;
   }
-  const {
-    title,
-    currentPrice,
-    price,
-    minimalBid,
-    type,
-    imageLinks,
-    views,
-    author,
-  } = product;
+  const { title, type, imageLinks, views, author } = product;
   const isAuction = type == ProductType.AUCTION;
 
+  const handleToggleFavorite = async (productId: string) => {
+    setIsLoading(true);
```

Comment:
You can use data status for detecting loading state

Reply:
fixed